packages/mui-joy/src/styles/extendTheme.ts

```diff
@@ -675,6 +677,7 @@ export default function extendTheme(themeOptions?: CssVarsThemeOptions): Theme {
     return createSpacing(spacing, createUnarySpacing(this));
   };
   theme.spacing = theme.generateSpacing();
+  theme.typography = mergedScales.typography as any; // cast to `any` to avoid internal module augmentation in the repo.
```

Comment:
Why is this redefined here as well as here: https://github.com/mui/material-ui/pull/41703/files#diff-158304f3a4b17f65c8bd9dd688681ae17e38aa6da3cabaaa8bbaae0a41948710R608?

Reply:
Let me clarify, it is redefined to preserve the same behavior as the developers passed the typography in.

In line 608, the typography is configured by `prepareTypographyVars` so that the generated token is in the format of `font: …`.

```js
prepareTypographyVars({ h1: { fontFamily: …, fontSize: …, lineHeight: …, fontWeight: … }})
// { h1: "…valid CSS `font` value", …other typography }
```

At the end, before the theme is return, the original `typography` object is redefined back so that this behavior remains the same:

```js
styled('div')({ theme }) => ({
  …theme.typography.h1,
})
```

Reply:
So, line 608 adds the `theme.vars.typography...` values?

Reply:
Not really, line 608 builds up the typography structure so that `prepareCssVars` at line 667 generates the `vars.typography` based on that structure.

Reply:
Thanks for the explanation!